lib/libc/string/bcopy.c

```diff
@@ -101,18 +104,19 @@ bcopy(const void *src0, void *dst0, size_t length)
 		 * Copy whole words, then mop up any trailing bytes.
 		 */
 		t = length / wsize;
-		TLOOP(*(word *)(void *)dst = *(const word *)(const void *)src;
+		TLOOP(*(word *)dst = *(const word *)src;
 		    src += wsize; dst += wsize);
 		t = length & wmask;
 		TLOOP(*dst++ = *src++);
+#ifndef MEMCOPY
```

Comment:
In practice, programs do depend on `memcpy()` working on overlapping areas sensibly, even though it is wrong. Therefore, breaking this is unwise.

At least on x86, I expect that the gain from not handling overlaps is quite small, except if modern CPUs have really lowered the setup cost of REP MOVSB so much that it can always be used. In such a case, small copies are best done by loading the data into registers and then storing it at the destination (taking advantage of unaligned access to reduce branches), which naturally handles overlaps, while the overhead of a branch is limited in large copies.

Reply:
Which programs would that be, if I may ask? Can I have an example?